src/util/math.h

```diff
@@ -65,6 +65,13 @@ inline int roundUpToPowerOf2(int v) {
     return power;
 }
 
+inline double roundToFraction(double value, int denominator) {
```

Comment:
we should round to the nearest value, otherwise all values have a "snaping region", but not the topmost value,
2 in this case, is only adopted on the very maximum.